x-pack/plugins/observability_solution/observability_ai_assistant/server/service/client/operators/continue_conversation.ts

```diff
@@ -148,6 +152,10 @@ function getFunctionDefinitions({
         [FunctionVisibility.AssistantOnly, FunctionVisibility.All].includes(def.visibility)
     );
 
+  if (typeof disableFunctions === 'object') {
+    systemFunctions = systemFunctions.filter((fn) => disableFunctions.except.includes(fn.name));
+  }
```

Comment:
I find this API weird. What about keeping `disableFunctions` as a boolean, then have a separate `allowedFunctions` that is an array of allowed functions. So by default all functions are included. With `disableFunctions=true` it is possible to not include any functions. And with `allowedFunctions=func1,func2` it's possible to cherry pick only select functions.

The `allowedFunctions` and `disableFunctions` should be mutually exclusive (although we don't have to enforce it).

Reply:
Agree that it is a little weird, but I use this structure precisely so we don't have to add any type or runtime guards around it. Plus we might end up wanting to disable _only_ specific functions (which would be `disableFunctions.only`). But maybe it's not worth it, WDYT?

Reply:
@sorenlouv I will just merge this in, in the interest of time but I'm happy to make this change separately if you feel it's worth it.